src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -181,6 +181,15 @@ private static void EndInvokeDotNetAfterTask(Task task, JSRuntime jsRuntime, in
 
                 throw;
             }
+            finally
+            {
+                // We require the invoked method to retrieve any pending byte arrays synchronously. If we didn't,
+                // we wouldn't be able to have overlapping async calls. As a way to enforce this, we clear the
+                // pending byte arrays synchronously after the call. This also helps because the recipient isn't
+                // required to consume all the pending byte arrays, since it's legal for the JS data model to contain
+                // more data than the .NET data model (like overposting)
+                jsRuntime.ByteArraysToBeRevived.Clear();
```

Comment:
Previously we cleared this set *before* invoking the user code via JS interop. But now we do it synchronously after. This is to allow the callee to retrieve byte arrays during custom deserialization. I don't think there are any drawbacks to this change.